### ISSUE TITLE
ACME/Cloudflare: Can't create wildcard cert for subdomain

### ISSUE THREAD

**Issue by MiniDigger** [containous/traefik#4200]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

trying to setup a new domain (*.some.subdomain.tld. some.subdomain.tld) with the cloudflare acme dns challenge provider. this is the first domain on this new instance, rules file is empty, it creates hostnames with docker labels.

### What did you expect to see?

traefik should generate a cert

### What did you see instead?

```
time="2018-11-15T19:58:23Z" level=info msg="legolog: [INFO] [*.some.subdomain.tld, some.subdomain.tld] acme: Obtaining bundled SAN certificate"
time="2018-11-15T19:58:24Z" level=info msg="legolog: [INFO] [*.some.subdomain.tld] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/******"
time="2018-11-15T19:58:24Z" level=info msg="legolog: [INFO] [some.subdomain.tld] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/******"
time="2018-11-15T19:58:24Z" level=info msg="legolog: [INFO] [some.subdomain.tld] acme: Could not find solver for: http-01"
time="2018-11-15T19:58:24Z" level=info msg="legolog: [INFO] [some.subdomain.tld] acme: Could not find solver for: tls-alpn-01"
time="2018-11-15T19:58:24Z" level=info msg="legolog: [INFO] [some.subdomain.tld] acme: Preparing to solve DNS-01"
time="2018-11-15T19:58:24Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
time="2018-11-15T19:58:24Z" level=error msg="Stack: goroutine 74 [running]:\nruntime/debug.Stack(0x265ea80, 0x17, 0xc00017a390)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/containous/traefik/safe.defaultRecoverGoroutine(0x21e5ca0, 0x4551d40)\n\t/go/src/github.com/containous/traefik/safe/routine.go:148 +0x89\ngithub.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc000749b68)\n\t/go/src/github.com/containous/traefik/safe/routine.go:156 +0x56\npanic(0x21e5ca0, 0x4551d40)\n\t/usr/local/go/src/runtime/panic.go:513 +0x1b9\ngithub.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/cloudflare.(*DNSProvider).Present(0xc0005f9b60, 0xc00073c540, 0x17, 0xc00045cb40, 0x2b, 0xc00088aa20, 0x57, 0x0, 0xc0007493f8)\n\t/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/providers/dns/cloudflare/cloudflare.go:134 +0x3ca\ngithub.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*dnsChallenge).PreSolve(0xc000920880, 0xc00008a380, 0x6a, 0xc0005327c0, 0x6, 0xc0005327c6, 0x7, 0xc00045cb40, 0x2b, 0x0, ...)\n\t/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:102 +0x17d\ngithub.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).solveChallengeForAuthz(0xc0001bc0c0, 0xc00013e540, 0x2, 0x2, 0x0, 0x0)\n\t/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:603 +0x3c6\ngithub.com/containous/traefik/vendor/github.com/xenolf/lego/acme.(*Client).ObtainCertificate(0xc0001bc0c0, 0xc0000e2a00, 0x2, 0x2, 0x4570501, 0x0, 0x0, 0xc00098bb00, 0x42b1f2, 0x8, ...)\n\t/go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/client.go:437 +0x2b3\ngithub.com/containous/traefik/provider/acme.obtainCertificateWithRetry.func1(0x8, 0x2714770)\n\t/go/src/github.com/containous/traefik/provider/acme/provider.go:487 +0x76\ngithub.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/containous/traefik/safe/routine.go:160 +0x62\ngithub.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc0005f9b80, 0x2951d20, 0xc0009208a0, 0x2714448, 0xc0002023c0, 0x17)\n\t/go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:37 +0xa2\ngithub.com/containous/traefik/provider/acme.obtainCertificateWithRetry(0xc0000e2a00, 0x2, 0x2, 0xc0001bc0c0, 0x1bf08eb000, 0x77359400, 0x1, 0x2, 0x2, 0x2)\n\t/go/src/github.com/containous/traefik/provider/acme/provider.go:501 +0x193\ngithub.com/containous/traefik/provider/acme.(*Provider).resolveCertificate(0xc0002a13b0, 0xc000202360, 0x19, 0xc00017bf80, 0x1, 0x1, 0xfad501, 0x0, 0x0, 0x0)\n\t/go/src/github.com/containous/traefik/provider/acme/provider.go:411 +0x3ba\ngithub.com/containous/traefik/provider/acme.(*Provider).Provide.func1()\n\t/go/src/github.com/containous/traefik/provider/acme/provider.go:186 +0x72\ngithub.com/containous/traefik/safe.GoWithRecover.func1(0x2714778, 0xc0000e2920)\n\t/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d\ncreated by github.com/containous/traefik/safe.GoWithRecover\n\t/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49\n"
time="2018-11-15T19:58:24Z" level=error msg="Error obtaining certificate retrying in 326.61237ms"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Traefik version v1.7.4 built on 2018-10-30_10:44:30AM
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http", "https"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

checkNewVersion = true

[acme]
#caServer = "https://acme-staging.api.letsencrypt.org/directory"
email = "********"
storage = "/acme.json"
entryPoint = "https"
acmeLogging = true
onDemand = true
onHostRule = true
  [acme.dnsChallenge]
  provider = "cloudflare"
  delayBeforeCheck = 0

[[acme.domains]]
  main = "*.some.subdomain.tld"
  sans = ["some.subdomain.tld"]

[api]
entryPoint = "traefik"
dashboard = true

[docker]
  endpoint = "unix:///var/run/docker.sock"
  domain = "docker.localhost"
  watch = true
  exposedbydefault = true
  usebindportip = true

[file]
  filename = "rules.toml"
watch = true
```

env:
CLOUDFLARE_EMAIL | ****
CLOUDFLARE_API_KEY | ****

its running on a ubuntu server 18.04.

**Comment by traefiker** [comment]
Closed by #4201.

**Comment by MiniDigger** [comment]
Thanks for the quick response!
